exp/internal/controllers/machinepool_controller.go

```diff
@@ -260,7 +261,9 @@ func (r *MachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *cl
 		return err
 	}
 
-	if err := r.reconcileDeleteNodes(ctx, cluster, mp); err != nil {
+	err := r.reconcileDeleteNodes(ctx, cluster, mp)
+	// Check if the return error has ClusterAccessorErrorMessage to ignore unreachable cluster.
+	if err != nil && !strings.Contains(err.Error(), remote.ClusterAccessorErrorMessage) {
```

Comment:
Should we implement a function like MachineReconciler's `isNodeDeleteAllowed` for MachinePool and call it before `reconcileDeleteNodes()`? I *think* that would be considered more graceful, but @chrischdi or @fabriziopandini please correct me if you had something else in mind.

Reply:
I'd favor that instead of relying on string matching on an error, yes 👍

Reply:
okay, that is fine. But the logic for deleting the machinePool if the cluster is unreachable Should we proceed with deleting the machinePool as the machinePool.infrastructureRef already doing validation for the node deletion OR we return and don't proceed with the delete ?

Reply:
Basically we should continue. The Machine controller works like this:

* Figure out if node deletion is "allowed" (allowed is not exactly perfect wording, but basically if we should do node deletion or not
* Result is stored in isDeleteNodeAllowed
* If isDeleteNodeAllowed is true
  * Try to delete Nodes
  * If this doesn't work either immediately or after some timeout (if NodeDeletionTimeout is configured) => log the error and continue with Machine deletion

I would expect the same logic for MachinePools. I think we should also handle NodeDeletionTimeout as this is also a field within MachinePools (MachinePool.spec.template.spec.nodeDeletionTimeout).



xrefs:
* https://github.com/kubernetes-sigs/cluster-api/blob/15615f64b8c8db57afb67384d57951f42d0887d3/internal/controllers/machine/machine_controller.go#L348
* https://github.com/kubernetes-sigs/cluster-api/blob/15615f64b8c8db57afb67384d57951f42d0887d3/internal/controllers/machine/machine_controller.go#L467-L488

Reply:
would you review this, I add the logic for machine Delete timeout